File: php/src/Types/Photo.php
<?php

namespace Authdog\Types;

/**
 * User photo
 */
class Photo
{
    public string $id;
    public string $value;
    public string $type;

    public function __construct(array $data)
    {
        $this->id = $data['id'] ?? '';
        $this->value = $data['value'] ?? '';
        $this->type = $data['type'] ?? '';
    }
}

File: planned/scala/src/main/scala/com/authdog/types/Photo.scala
package com.authdog.types

import io.circe.{Decoder, Encoder, HCursor, Json}
import io.circe.syntax._

/**
 * User photo
 */
case class Photo(
  id: String,
  value: String,
  `type`: String
)

object Photo {
  implicit val decoder: Decoder[Photo] = (c: HCursor) => for {
    id <- c.downField("id").as[String]
    value <- c.downField("value").as[String]
    `type` <- c.downField("type").as[String]
  } yield Photo(id, value, `type`)
  
  implicit val encoder: Encoder[Photo] = (photo: Photo) => Json.obj(
    "id" -> photo.id.asJson,
    "value" -> photo.value.asJson,
    "type" -> photo.`type`.asJson
  )
}
